paraatm/simulation_method/integrated_flight_health_monitoring.py

```diff
@@ -0,0 +1,302 @@
+"""
+NASA University Leadership Initiative program
+Information fusion for real-time national air transportation system prognostics under uncertainty
+
+    Integrated Flight Heatlrh Monitoring (IFHM) module
+    - Simulate normal flight operation
+    - Simulate flight with upset flight scenarios using 1) upgen sub module
+    - Monitor flight health in real-time and detect aircraft upsets
+      using 2) fhm sub module
+    - Generate output files to perform analysis
+
+    @author: Hyunseong Lee, Adaptive Intelligent Materials & Systems (AIMS) Center,
+             Arizona State University
+    
+    Last modified on 5/25/2020
+"""
+
+import time
+import os
+from jpype import *
+from array import *
+from shutil import copyfile
+from scipy import signal
+import numpy as np
+import math
+import pandas as pd
+import tensorflow as tf
+from tensorflow.keras import backend
+from tensorflow.keras.models import Model, load_model
+from tensorflow.keras.layers import Input, Dense, BatchNormalization
+from tensorflow.keras.regularizers import l1
+from tensorflow.keras import optimizers
+from paraatm.io.gnats import GnatsSimulationWrapper
+from paraatm.simulation_method.upset_gen import upgen
+from paraatm.simulation_method.flight_health_monitor import fhm
+
+class IFHM(GnatsSimulationWrapper):
+    def __init__(self, sim_inputs):
+        
+        # GNATS simulation input files
+        self.fp_file = sim_inputs["fp_file"]
+        self.mfl_file = sim_inputs["mfl_file"]
+        self.cruz_alt = sim_inputs["cruz_alt"]
+        self.upset_case = sim_inputs["upset_case"]
+        self.upset_init = sim_inputs["upset_init"]
+        self.upset_dur = sim_inputs["upset_dur"]
+        self.alt_rate_coef = sim_inputs["alt_rate_coef"]
+        self.tas_rate_coef = sim_inputs["tas_rate_coef"]
+        self.course_rate_coef = sim_inputs["course_rate_coef"]
+        self.th_Mahal_dist = sim_inputs["th_Mahal_dist"]
+        
+        # Define GNATS-Server environment (not standalone) & perform simulation
+        # Set PARA-ATM & GNATS folders
+        # PARA_ATM_Home = os.environ.get('PARA_ATM_Home')
+        GNATS_Home = os.environ.get("GNATS_Home")
+        GNATS_Client = GNATS_Home + "/GNATS_Client"
+        GNATS_Server = GNATS_Home + "/GNATS_Server"
+        
+        # Locate working directory to GNATS Client folder
+        os.chdir(GNATS_Client)
+        
+        # Set classpath & start JVM (GNATS-Server)
+        classpath = "dist/gnats-client.jar:dist/gnats-shared.jar:dist/json.jar:dist/rmiio-2.1.2.jar:dist/commons-logging-1.2.jar"
+        startJVM(getDefaultJVMPath(), "-ea", "-Djava.class.path=%s" % classpath)
```

Comment:
This can all be replaced by a call to `GnatsEnvironment.start_jvm`, which comes from [paraatm/io/gnats.py](../blob/master/paraatm/io/gnats.py).  In addition to doing this for you, it prevents multiple calls to `jpype.startJVM`, which would otherwise cause a crash (as written, your `__init__` function cannot be called more than once).  `GnatsEnvironment` will also handle automatically stopping the GNATS standalone, stopping the JVM, and restoring your original working directory.